src/app/auth/auth-direct-login/auth-direct-login.component.ts

```diff
@@ -122,6 +125,7 @@ export class AuthDirectLoginComponent implements OnInit {
             if (!this.utils.isEmpty(res) && res.status === 'forbidden' && [
               'User is not registered'
             ].includes(res.data.message)) {
+              this.storage.set('directLinkData', this._redirect(true));
```

Comment:
The logic to set the redirect route into local storage should be part of the function `_redirect()`
So `_redirect()` will either save the route to local storage for later redirection or redirect directly.
Given that, we can create a function called `_saveOrRedirect(route: Array, save = false)`, so that we can call this in the switch/case branches inside `_redirect()`